### PR TITLE
feat: Introduction to Smart Contracts 

### LINKED ISSUE #142: feat: Introduction to Smart Contracts
The Cairo Book must be beginner friendly - we do not expect prior knowledge of blockchain programming languages, and we should provide some explanation about what smart contracts are, what they can be used for, and why they're not the same on Starknet as on Ethereum (mostly, why are we using Cairo ?)

This part might be overlapping with the [Starknet Book](https://book.starknet.io/chapter_2/index.html) - But I think it's still great to have a succinct introduction in our book.

Estimated time: 2 to 3 hours

### PR DESCRIPTION
**Pull request by 0xEniotna**
Should close #142

### CODE REVIEW

File: src/ch99-01-01-introduction-to-smart-contracts.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+The programming language used to write smart contracts varies depending on the blockchain. For example, on Ethereum and the [EVM-compatible ecosystem](https://ethereum.org/en/developers/docs/evm/) ecosystem, the most commonly used language is Solidity, while on Starknet, it is Cairo. The way the code is compiled also differs based on the blockchain. On Ethereum, Solidity is compiled into bytecode. On Starknet, Cairo is compiled into Sierra and then into Cairo Assembly (casm).
+
+Smart contracts possess several unique characteristics. They are **permissionless**, meaning anyone can deploy a smart contract on the network (within the context of a decentralized blockchain, of course). Smart contracts are also **transparent**; the data stored by the smart contract is accessible to anyone. The code that composes the contract can also be transparent, enabling **composability**. This allows developers to write smart contracts that utilize other smart contracts. Smart contracts are only **aware of themselves** and their immediate environment, requiring third-party software to access external data.
```
[the remark this answers is not in the document]
Comment (enitrat) [reply src/ch99-01-01-introduction-to-smart-contracts.md:12]: > Smart contracts are only aware of themselves and their immediate environment, requiring third-party software to access external data.

I think it's not explicit enough. Did you mean that smart-contracts only have access to data related to the blockchain they're deployed on, and cannot access anything from the "outside" world?

[the remark this answers is not in the document]
Comment (enitrat) [reply src/ch99-01-01-introduction-to-smart-contracts.md:12]: > Bruh, so replacing "use" by "utilize" is what GPT means by "The syntax is mostly correct, but there are a few small adjustments that can be made to improve clarity and readability:".

This is the most effective way of knowing whether something was written by ChatGPT or not, lol. instant detection and good hitrate.

> yes, that's what I meant. I kinda like using "only aware of themselves/ seulement conscient d'eux-mêmes" to describe that

 Can you think of another wording to make it more obvious pls?

File: src/ch99-01-01-introduction-to-smart-contracts.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
...n the blockchain space states that it is impossible to achieve a high level of scalability, decentralization, and security simultaneously; trade-offs must be made. Ethereum is at the intersection of decentralization and security. Eventually, it was decided that Ethereum's purpose would be to serve as a secure settlement layer, while complex computations would be offloaded to other networks built on top of Ethereum. These are called Layer 2s (L2s). The two primary types of L2s are optimistic rollups and validity rollups. Both approaches involve compressing and batching numerous transactions together, computing the new state, and settling the result on Ethereum (L1). The difference lies in the way the result is settled on L1. For optimistic rollups, the new state is considered valid by default, but there is a 7-day window for nodes to identify malicious transactions. In contrast, validity rollups, such as Starknet, use cryptography to prove that the new state has been correctly computed. This is the purpose of STARKs. You can learn more about STARKs from Starkware's Medium [article](https://medium.com/starkware/starks-starkex-and-starknet-9a426680745a), which serves as a good primer.
```
Comment (fricoben) [inline src/ch99-01-01-introduction-to-smart-contracts.md:44]: ``This is the purpose of STARKs, this cryptographic technology could permit validity rollups to scale significantly more than optimistic rollups.``

Let's make them understand why we choose ZK haha

File: src/ch99-01-01-introduction-to-smart-contracts.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+Remember Cairo? It is, in fact, a language developed specifically to work with STARKs and make them general-purpose. With Cairo, we can write **provable code**. In the context of Starknet, this allows proving the correctness of computations from one state to another. Unlike most (if not all) of Starknet's competitors that chose to use the EVM (either as-is or adapted) as a base layer, Starknet employs its own VM. This frees developers from the constraints of the EVM, opening up a broader range of possibilities. Coupled with decreased transaction costs, the combination of Starknet and Cairo creates an exiting playground for developers. Native account abstraction enables more complex logic for accounts and transaction flows. Emerging use cases include **transparent AI** and artificial intelligence and machine learning applications. Finally, **blockchain games** can be developed entirely **on-chain**.
```
Comment (fricoben) [inline src/ch99-01-01-introduction-to-smart-contracts.md:46]: After `This frees developers from the constraints of the EVM, opening up a broader range of possibilities. Coupled with decreased transaction costs, the combination of Starknet and Cairo creates an exiting playground for developers. `

I would add 

`Starknet has been specifically designed to maximize the use of STARK proofs for optimal scalability, Starknet is intentionally designed to maximize the capabilities of STARK proofs for optimal scalability, even if it means giving up EVM compatibility.`

**Review by fricoben — APPROVED** [review]
Nice PR ! Some comments.

**Review by enitrat — APPROVED** [review]

**Comment by enitrat** [comment]
@fricoben good?

**Review by fricoben — APPROVED** [review]
lgtm

**Pull request merged**